libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -379,7 +380,7 @@ internal class MapboxTripSession(
         }
 
         private fun calculateRemainingWaypoints(tripStatus: TripStatus): Int {
-            val routeCoordinates = tripStatus.route?.routeOptions?.coordinatesList()
+            val routeCoordinates = tripStatus.route?._waypoints()
```

Comment:
This would now also report silent waypoints which is a breaking change. It should only report regular and EV waypoints, so actually the ones that create legs.

Reply:
why it's breaking changes? We haven't had EV waypoints, as they appeared, now is a part of the API if anybody uses it

Reply:
I'm talking about silent waypoints. They are not really waypoints and should not be part of the remaining waypoints count.

Reply:
good catch 👍

Reply:
Also, silent and Ev waypoints are under `NavigationStatus#getNextWaypointIndex`(which is root of the `remainingWaypoints` value)

Reply:
I think it's rather confusing that we call this a `remainingWaypoint` when a silent waypoint is not truly a waypoint in the Directions API terminology.

I'm worried that if someone used to use `remainingWaypoints` to compare it with `RouteOptions#coordinates` (as that's the only practical usage because `remainingWaypoints` contains silent waypoints) then the proposed change will break that logic.

Reply:
from the SDK side, it seems like not a breaking change, it might be a misunderstanding between the SDK docs and customers. Let's considered it in the API doc and add the warning to the changelog. WDYT?

Reply:
Yea, this is fine, refs https://github.com/mapbox/mapbox-navigation-android/pull/6005#issuecomment-1292301010. But I agree that we should add a note in the docs that `remainingWaypoints` now might include more than requested waypoints if EV is used.